data-prepper-core/src/main/java/org/opensearch/dataprepper/plugin/PluginCreator.java

```diff
@@ -36,6 +48,8 @@ <T> T newPluginInstance(final Class<T> pluginClass,
 
         final Object[] constructorArguments = pluginArgumentsContext.createArguments(constructor.getParameterTypes());
 
+        registerPluginConfigurationObservables(constructorArguments);
```

Comment:
This logic is not related to the logic of creating a new instance of a plugin. This change is thus mixing concerns which should remain separate.

What might be a better design would be to split this existing logic into multiple classes. The existing `PluginCreator` class can still coordinate all the work. But, the logic of instantiation and even of extracting arguments can be split into other classes.